### ISSUE TITLE
directory should be able to contain users of multiple identity types

### ISSUE THREAD

**Issue by adobeDan** [adobe-apiplatform/user-sync.py#48]
Customers with centralized user directories should be able to have users of different types in those directories (e.g., different domains some of which are enterpriseID and some of which are federatedID, or adobeID users mixed in with enterpriseID users).

To support this, we should get the identity type from a directory attribute, in the same way we use the `type` column in CSV input.  There is already a customization mechanism for customers to pick the attribute (and format) used for email address info in the directory; that mechanism should be extended to support username, domain, and identity type.

This would override the `user_identity_type` option at both the connector and global levels.

**Comment by phil-levy** [comment]
Is this task mainly adding the ability in the customization to set the identity type?  Or are you thinking of adding more declarative customization in the connector-ldap config file?  Seems low use enough that we could live with the former for a long time.

**Comment by adobeDan** [comment]
So there is supposed to be a new-user type override possible in connector-ldap that applies to everyone read from the directory (it's an undocumented field in the config called `user_identity_type`), but I don't think it's actually implemented properly.  But having a per-connector override isn't too useful, because every sync run only uses one connector, so you might as well just set the `new_account_type` in the main config.

What I'm talking about is being able to set the identity type of a specific person in your directory (the same way you can in a csv file).  Obviously you'd have to have the same setting for all users in a given domain, but directories can have multiple domains in them not all of which are mapped to the same directory on the Adobe side.

So basically I'm just talking about doing the same thing for directory input that we already support for CSV input.  Does that answer your question?  (It's actually a very small change.)
